pkg/storage/allocator.go

```diff
@@ -541,7 +541,8 @@ func filterBehindReplicas(
 	candidates := make([]roachpb.ReplicaDescriptor, 0, len(replicas))
 	for _, r := range replicas {
 		if progress, ok := raftStatus.Progress[uint64(r.ReplicaID)]; ok {
-			if progress.State == raft.ProgressStateReplicate &&
+			if (progress.State == raft.ProgressStateReplicate ||
+				progress.State == raft.ProgressStateProbe) &&
```

Comment:
Thanks, it didn't look quite right. This makes the test a bit fugly, but appears to work in practice. Thanks.